pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -148,12 +177,56 @@ func (c *globalIngressIPController) onDelete(ingressIP *submarinerv1.GlobalIngre
 
 	key, _ := cache.MetaNamespaceKeyFunc(ingressIP)
 
+	if ingressIP.Spec.ServiceRef != nil {
```

Comment:
`ingressIP.Spec.ServiceRef` isn't used in this block - did you mean to check `ingressIP.Spec.Target`?

```suggestion
	if ingressIP.Spec.Target == submarinerv1.ClusterIPService {
```

Reply:
Checking either one will work.